File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.js
function solution(x) {
    
    let str = x.toString();
    
    let sum = 0;
    
    for(const s of str){
        sum += Number(s);
    }
    
    if(x % sum === 0){
        return true;
    }else {
        return false;
    }
    
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
import java.util.*;

class Solution {
    public boolean solution(int x) {
    
        String x_str = x+"";
        
        int sum = 0;
        
        for(int i = 0; i < x_str.length(); i++){
            sum += (x_str.charAt(i)-'0');
        }
        
        boolean check = (x % sum == 0) ? true : false;
        
        return check;
    }
}
